e2e/tests/LoggedIn/livePublication.e2e.spec.ts

```diff
@@ -127,16 +130,28 @@ test.describe('Live Publication', () => {
         await expect(page.locator(PageModel.authorInfo.result)).toBeVisible();
     });
 
-    test.skip('Download pdf/json', async ({ browser }) => {
-        // test TODO
-        // Start up test
+    test('Download pdf/json', async ({ browser, headless }) => {
         const page = await browser.newPage();
-
-        // Login
         await page.goto(Helpers.UI_BASE);
         await Helpers.login(page, browser);
         await expect(page.locator(PageModel.header.usernameButton)).toHaveText(`${Helpers.user1.fullName}`);
         await page.goto(`${Helpers.UI_BASE}/publications/cl3fz14dr0001es6i5ji51rq4`, { waitUntil: 'domcontentloaded' });
+
+        // Behaviour is different depending on whether test is running headless or not.
+        if (headless) {
+            // Download PDF
+            const downloadPromise = page.waitForEvent('download');
+            await page.locator('aside').getByLabel('Download PDF').click();
+            const download = await downloadPromise;
```

Comment:
Curious, in my experience with these kind of events I've needed to await both promises at the same time in a promise.all, otherwise there's a chance the download event fires before you start awaiting the download event.

Is this showing flaky behaviour at all?

Reply:
Of course playwright are being inconsistent here.. Elsewhere they recommend doing a promise.all when waiting for events.

It might be ok then if playwright suggest it this way.